packages/grafana-ui/src/components/Table/TableNG/TableNG.tsx

```diff
@@ -241,16 +240,12 @@ export function TableNG(props: TableNGProps) {
     return props.data.fields.length;
   }, [props.data.fields]);
 
-  const fieldDisplayType = useMemo(() => {
-    return props.data.fields.reduce(
-      (acc, { config, name }) => {
-        if (config?.custom?.cellOptions?.type) {
-          acc[name] = config.custom.cellOptions.type;
-        }
-        return acc;
-      },
-      {} as Record<string, TableCellDisplayMode>
-    );
+  const fieldDisplayType: Record<string, TableCellDisplayMode> = useMemo(() => {
+    return props.data.fields.reduce((acc, { config, name }) => {
+      // @ts-ignore
+      acc[name] = config.custom.cellOptions.type;
```

Comment:
@ashharrison90 can we map string back to enum value somehow?

Reply:
no, but i don't think you need to, you can just type the `.reduce` properly. move your `Record<string, TableCellDisplayMode>` to be a generic on the `.reduce`:
```tsx
  const fieldDisplayType  = useMemo(() => {
    return props.data.fields.reduce<Record<string, TableCellDisplayMode>>((acc, { config, name }) => {
      acc[name] = config.custom.cellOptions.type;
```